// @aws-sdk/client-api-gateway/dist-es/commands/CreateRequestValidatorCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateRequestValidatorRequestFilterSensitiveLog, RequestValidatorFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1CreateRequestValidatorCommand, serializeAws_restJson1CreateRequestValidatorCommand, } from "../protocols/Aws_restJson1";
var CreateRequestValidatorCommand = (function (_super) {
    __extends(CreateRequestValidatorCommand, _super);
    function CreateRequestValidatorCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateRequestValidatorCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "CreateRequestValidatorCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateRequestValidatorRequestFilterSensitiveLog,
            outputFilterSensitiveLog: RequestValidatorFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateRequestValidatorCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateRequestValidatorCommand(input, context);
    };
    CreateRequestValidatorCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateRequestValidatorCommand(output, context);
    };
    return CreateRequestValidatorCommand;
}($Command));
export { CreateRequestValidatorCommand };
