// @aws-sdk/client-api-gateway/dist-es/commands/CreateBasePathMappingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BasePathMappingFilterSensitiveLog, CreateBasePathMappingRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1CreateBasePathMappingCommand, serializeAws_restJson1CreateBasePathMappingCommand, } from "../protocols/Aws_restJson1";
var CreateBasePathMappingCommand = (function (_super) {
    __extends(CreateBasePathMappingCommand, _super);
    function CreateBasePathMappingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateBasePathMappingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "CreateBasePathMappingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateBasePathMappingRequestFilterSensitiveLog,
            outputFilterSensitiveLog: BasePathMappingFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateBasePathMappingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateBasePathMappingCommand(input, context);
    };
    CreateBasePathMappingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateBasePathMappingCommand(output, context);
    };
    return CreateBasePathMappingCommand;
}($Command));
export { CreateBasePathMappingCommand };
