// @aws-sdk/client-api-gateway/dist-es/commands/CreateUsagePlanKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateUsagePlanKeyRequestFilterSensitiveLog, UsagePlanKeyFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1CreateUsagePlanKeyCommand, serializeAws_restJson1CreateUsagePlanKeyCommand, } from "../protocols/Aws_restJson1";
var CreateUsagePlanKeyCommand = (function (_super) {
    __extends(CreateUsagePlanKeyCommand, _super);
    function CreateUsagePlanKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateUsagePlanKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "CreateUsagePlanKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateUsagePlanKeyRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UsagePlanKeyFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateUsagePlanKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateUsagePlanKeyCommand(input, context);
    };
    CreateUsagePlanKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateUsagePlanKeyCommand(output, context);
    };
    return CreateUsagePlanKeyCommand;
}($Command));
export { CreateUsagePlanKeyCommand };
