// @aws-sdk/client-api-gateway/dist-es/commands/CreateDocumentationPartCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateDocumentationPartRequestFilterSensitiveLog, DocumentationPartFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1CreateDocumentationPartCommand, serializeAws_restJson1CreateDocumentationPartCommand, } from "../protocols/Aws_restJson1";
var CreateDocumentationPartCommand = (function (_super) {
    __extends(CreateDocumentationPartCommand, _super);
    function CreateDocumentationPartCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateDocumentationPartCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "CreateDocumentationPartCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDocumentationPartRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DocumentationPartFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDocumentationPartCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateDocumentationPartCommand(input, context);
    };
    CreateDocumentationPartCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateDocumentationPartCommand(output, context);
    };
    return CreateDocumentationPartCommand;
}($Command));
export { CreateDocumentationPartCommand };
