// @aws-sdk/client-api-gateway/dist-es/commands/CreateStageCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateStageRequestFilterSensitiveLog, StageFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1CreateStageCommand, serializeAws_restJson1CreateStageCommand, } from "../protocols/Aws_restJson1";
var CreateStageCommand = (function (_super) {
    __extends(CreateStageCommand, _super);
    function CreateStageCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateStageCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "CreateStageCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateStageRequestFilterSensitiveLog,
            outputFilterSensitiveLog: StageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateStageCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateStageCommand(input, context);
    };
    CreateStageCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateStageCommand(output, context);
    };
    return CreateStageCommand;
}($Command));
export { CreateStageCommand };
