// @aws-sdk/client-api-gateway/dist-es/commands/CreateUsagePlanCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateUsagePlanRequestFilterSensitiveLog, UsagePlanFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1CreateUsagePlanCommand, serializeAws_restJson1CreateUsagePlanCommand, } from "../protocols/Aws_restJson1";
var CreateUsagePlanCommand = (function (_super) {
    __extends(CreateUsagePlanCommand, _super);
    function CreateUsagePlanCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateUsagePlanCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "CreateUsagePlanCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateUsagePlanRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UsagePlanFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateUsagePlanCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateUsagePlanCommand(input, context);
    };
    CreateUsagePlanCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateUsagePlanCommand(output, context);
    };
    return CreateUsagePlanCommand;
}($Command));
export { CreateUsagePlanCommand };
