// @aws-sdk/client-api-gateway/dist-es/commands/DeleteDocumentationPartCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDocumentationPartRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1DeleteDocumentationPartCommand, serializeAws_restJson1DeleteDocumentationPartCommand, } from "../protocols/Aws_restJson1";
var DeleteDocumentationPartCommand = (function (_super) {
    __extends(DeleteDocumentationPartCommand, _super);
    function DeleteDocumentationPartCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDocumentationPartCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "DeleteDocumentationPartCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDocumentationPartRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDocumentationPartCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteDocumentationPartCommand(input, context);
    };
    DeleteDocumentationPartCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteDocumentationPartCommand(output, context);
    };
    return DeleteDocumentationPartCommand;
}($Command));
export { DeleteDocumentationPartCommand };
