// @aws-sdk/client-api-gateway/dist-es/commands/DeleteRequestValidatorCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteRequestValidatorRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1DeleteRequestValidatorCommand, serializeAws_restJson1DeleteRequestValidatorCommand, } from "../protocols/Aws_restJson1";
var DeleteRequestValidatorCommand = (function (_super) {
    __extends(DeleteRequestValidatorCommand, _super);
    function DeleteRequestValidatorCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteRequestValidatorCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "DeleteRequestValidatorCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteRequestValidatorRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteRequestValidatorCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteRequestValidatorCommand(input, context);
    };
    DeleteRequestValidatorCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteRequestValidatorCommand(output, context);
    };
    return DeleteRequestValidatorCommand;
}($Command));
export { DeleteRequestValidatorCommand };
