// @aws-sdk/client-api-gateway/dist-es/commands/DeleteModelCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteModelRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1DeleteModelCommand, serializeAws_restJson1DeleteModelCommand, } from "../protocols/Aws_restJson1";
var DeleteModelCommand = (function (_super) {
    __extends(DeleteModelCommand, _super);
    function DeleteModelCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteModelCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "DeleteModelCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteModelRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteModelCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteModelCommand(input, context);
    };
    DeleteModelCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteModelCommand(output, context);
    };
    return DeleteModelCommand;
}($Command));
export { DeleteModelCommand };
