// @aws-sdk/client-api-gateway/dist-es/commands/DeleteMethodResponseCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteMethodResponseRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1DeleteMethodResponseCommand, serializeAws_restJson1DeleteMethodResponseCommand, } from "../protocols/Aws_restJson1";
var DeleteMethodResponseCommand = (function (_super) {
    __extends(DeleteMethodResponseCommand, _super);
    function DeleteMethodResponseCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteMethodResponseCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "DeleteMethodResponseCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteMethodResponseRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteMethodResponseCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteMethodResponseCommand(input, context);
    };
    DeleteMethodResponseCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteMethodResponseCommand(output, context);
    };
    return DeleteMethodResponseCommand;
}($Command));
export { DeleteMethodResponseCommand };
