// @aws-sdk/client-api-gateway/dist-es/commands/DeleteBasePathMappingCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteBasePathMappingRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1DeleteBasePathMappingCommand, serializeAws_restJson1DeleteBasePathMappingCommand, } from "../protocols/Aws_restJson1";
var DeleteBasePathMappingCommand = (function (_super) {
    __extends(DeleteBasePathMappingCommand, _super);
    function DeleteBasePathMappingCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteBasePathMappingCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "DeleteBasePathMappingCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteBasePathMappingRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteBasePathMappingCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteBasePathMappingCommand(input, context);
    };
    DeleteBasePathMappingCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteBasePathMappingCommand(output, context);
    };
    return DeleteBasePathMappingCommand;
}($Command));
export { DeleteBasePathMappingCommand };
