// @aws-sdk/client-api-gateway/dist-es/commands/DeleteApiKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteApiKeyRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1DeleteApiKeyCommand, serializeAws_restJson1DeleteApiKeyCommand, } from "../protocols/Aws_restJson1";
var DeleteApiKeyCommand = (function (_super) {
    __extends(DeleteApiKeyCommand, _super);
    function DeleteApiKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteApiKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "DeleteApiKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteApiKeyRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteApiKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteApiKeyCommand(input, context);
    };
    DeleteApiKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteApiKeyCommand(output, context);
    };
    return DeleteApiKeyCommand;
}($Command));
export { DeleteApiKeyCommand };
