// @aws-sdk/client-api-gateway/dist-es/commands/DeleteIntegrationResponseCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteIntegrationResponseRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1DeleteIntegrationResponseCommand, serializeAws_restJson1DeleteIntegrationResponseCommand, } from "../protocols/Aws_restJson1";
var DeleteIntegrationResponseCommand = (function (_super) {
    __extends(DeleteIntegrationResponseCommand, _super);
    function DeleteIntegrationResponseCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteIntegrationResponseCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "DeleteIntegrationResponseCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteIntegrationResponseRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteIntegrationResponseCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteIntegrationResponseCommand(input, context);
    };
    DeleteIntegrationResponseCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteIntegrationResponseCommand(output, context);
    };
    return DeleteIntegrationResponseCommand;
}($Command));
export { DeleteIntegrationResponseCommand };
