// @aws-sdk/client-api-gateway/dist-es/commands/DeleteDeploymentCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDeploymentRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1DeleteDeploymentCommand, serializeAws_restJson1DeleteDeploymentCommand, } from "../protocols/Aws_restJson1";
var DeleteDeploymentCommand = (function (_super) {
    __extends(DeleteDeploymentCommand, _super);
    function DeleteDeploymentCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDeploymentCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "DeleteDeploymentCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDeploymentRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDeploymentCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteDeploymentCommand(input, context);
    };
    DeleteDeploymentCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteDeploymentCommand(output, context);
    };
    return DeleteDeploymentCommand;
}($Command));
export { DeleteDeploymentCommand };
