// @aws-sdk/client-api-gateway/dist-es/commands/FlushStageCacheCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { FlushStageCacheRequestFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1FlushStageCacheCommand, serializeAws_restJson1FlushStageCacheCommand, } from "../protocols/Aws_restJson1";
var FlushStageCacheCommand = (function (_super) {
    __extends(FlushStageCacheCommand, _super);
    function FlushStageCacheCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    FlushStageCacheCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "FlushStageCacheCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: FlushStageCacheRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    FlushStageCacheCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1FlushStageCacheCommand(input, context);
    };
    FlushStageCacheCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1FlushStageCacheCommand(output, context);
    };
    return FlushStageCacheCommand;
}($Command));
export { FlushStageCacheCommand };
