// @aws-sdk/client-api-gateway/dist-es/commands/FlushStageAuthorizersCacheCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { FlushStageAuthorizersCacheRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1FlushStageAuthorizersCacheCommand, serializeAws_restJson1FlushStageAuthorizersCacheCommand, } from "../protocols/Aws_restJson1";
var FlushStageAuthorizersCacheCommand = (function (_super) {
    __extends(FlushStageAuthorizersCacheCommand, _super);
    function FlushStageAuthorizersCacheCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    FlushStageAuthorizersCacheCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "FlushStageAuthorizersCacheCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: FlushStageAuthorizersCacheRequestFilterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    FlushStageAuthorizersCacheCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1FlushStageAuthorizersCacheCommand(input, context);
    };
    FlushStageAuthorizersCacheCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1FlushStageAuthorizersCacheCommand(output, context);
    };
    return FlushStageAuthorizersCacheCommand;
}($Command));
export { FlushStageAuthorizersCacheCommand };
