// @aws-sdk/client-api-gateway/dist-es/commands/GenerateClientCertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ClientCertificateFilterSensitiveLog, GenerateClientCertificateRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GenerateClientCertificateCommand, serializeAws_restJson1GenerateClientCertificateCommand, } from "../protocols/Aws_restJson1";
var GenerateClientCertificateCommand = (function (_super) {
    __extends(GenerateClientCertificateCommand, _super);
    function GenerateClientCertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GenerateClientCertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GenerateClientCertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GenerateClientCertificateRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ClientCertificateFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GenerateClientCertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GenerateClientCertificateCommand(input, context);
    };
    GenerateClientCertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GenerateClientCertificateCommand(output, context);
    };
    return GenerateClientCertificateCommand;
}($Command));
export { GenerateClientCertificateCommand };
