// @aws-sdk/client-api-gateway/dist-es/commands/GetApiKeysCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ApiKeysFilterSensitiveLog, GetApiKeysRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetApiKeysCommand, serializeAws_restJson1GetApiKeysCommand, } from "../protocols/Aws_restJson1";
var GetApiKeysCommand = (function (_super) {
    __extends(GetApiKeysCommand, _super);
    function GetApiKeysCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetApiKeysCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetApiKeysCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetApiKeysRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ApiKeysFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetApiKeysCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetApiKeysCommand(input, context);
    };
    GetApiKeysCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetApiKeysCommand(output, context);
    };
    return GetApiKeysCommand;
}($Command));
export { GetApiKeysCommand };
