// @aws-sdk/client-api-gateway/dist-es/commands/GetApiKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ApiKeyFilterSensitiveLog, GetApiKeyRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetApiKeyCommand, serializeAws_restJson1GetApiKeyCommand, } from "../protocols/Aws_restJson1";
var GetApiKeyCommand = (function (_super) {
    __extends(GetApiKeyCommand, _super);
    function GetApiKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetApiKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetApiKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetApiKeyRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ApiKeyFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetApiKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetApiKeyCommand(input, context);
    };
    GetApiKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetApiKeyCommand(output, context);
    };
    return GetApiKeyCommand;
}($Command));
export { GetApiKeyCommand };
