// @aws-sdk/client-api-gateway/dist-es/commands/GetBasePathMappingsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BasePathMappingsFilterSensitiveLog, GetBasePathMappingsRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetBasePathMappingsCommand, serializeAws_restJson1GetBasePathMappingsCommand, } from "../protocols/Aws_restJson1";
var GetBasePathMappingsCommand = (function (_super) {
    __extends(GetBasePathMappingsCommand, _super);
    function GetBasePathMappingsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetBasePathMappingsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetBasePathMappingsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetBasePathMappingsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: BasePathMappingsFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetBasePathMappingsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetBasePathMappingsCommand(input, context);
    };
    GetBasePathMappingsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetBasePathMappingsCommand(output, context);
    };
    return GetBasePathMappingsCommand;
}($Command));
export { GetBasePathMappingsCommand };
