// @aws-sdk/client-api-gateway/dist-es/commands/GetDeploymentsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeploymentsFilterSensitiveLog, GetDeploymentsRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetDeploymentsCommand, serializeAws_restJson1GetDeploymentsCommand, } from "../protocols/Aws_restJson1";
var GetDeploymentsCommand = (function (_super) {
    __extends(GetDeploymentsCommand, _super);
    function GetDeploymentsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetDeploymentsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetDeploymentsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetDeploymentsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DeploymentsFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetDeploymentsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetDeploymentsCommand(input, context);
    };
    GetDeploymentsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetDeploymentsCommand(output, context);
    };
    return GetDeploymentsCommand;
}($Command));
export { GetDeploymentsCommand };
