// @aws-sdk/client-api-gateway/dist-es/commands/GetDocumentationVersionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DocumentationVersionsFilterSensitiveLog, GetDocumentationVersionsRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetDocumentationVersionsCommand, serializeAws_restJson1GetDocumentationVersionsCommand, } from "../protocols/Aws_restJson1";
var GetDocumentationVersionsCommand = (function (_super) {
    __extends(GetDocumentationVersionsCommand, _super);
    function GetDocumentationVersionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetDocumentationVersionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetDocumentationVersionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetDocumentationVersionsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DocumentationVersionsFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetDocumentationVersionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetDocumentationVersionsCommand(input, context);
    };
    GetDocumentationVersionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetDocumentationVersionsCommand(output, context);
    };
    return GetDocumentationVersionsCommand;
}($Command));
export { GetDocumentationVersionsCommand };
