// @aws-sdk/client-api-gateway/dist-es/commands/GetExportCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ExportResponseFilterSensitiveLog, GetExportRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetExportCommand, serializeAws_restJson1GetExportCommand, } from "../protocols/Aws_restJson1";
var GetExportCommand = (function (_super) {
    __extends(GetExportCommand, _super);
    function GetExportCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetExportCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetExportCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetExportRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ExportResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetExportCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetExportCommand(input, context);
    };
    GetExportCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetExportCommand(output, context);
    };
    return GetExportCommand;
}($Command));
export { GetExportCommand };
