// @aws-sdk/client-api-gateway/dist-es/commands/GetGatewayResponsesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GatewayResponsesFilterSensitiveLog, GetGatewayResponsesRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetGatewayResponsesCommand, serializeAws_restJson1GetGatewayResponsesCommand, } from "../protocols/Aws_restJson1";
var GetGatewayResponsesCommand = (function (_super) {
    __extends(GetGatewayResponsesCommand, _super);
    function GetGatewayResponsesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetGatewayResponsesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetGatewayResponsesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetGatewayResponsesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: GatewayResponsesFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetGatewayResponsesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetGatewayResponsesCommand(input, context);
    };
    GetGatewayResponsesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetGatewayResponsesCommand(output, context);
    };
    return GetGatewayResponsesCommand;
}($Command));
export { GetGatewayResponsesCommand };
