// @aws-sdk/client-api-gateway/dist-es/commands/GetIntegrationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetIntegrationRequestFilterSensitiveLog, IntegrationFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetIntegrationCommand, serializeAws_restJson1GetIntegrationCommand, } from "../protocols/Aws_restJson1";
var GetIntegrationCommand = (function (_super) {
    __extends(GetIntegrationCommand, _super);
    function GetIntegrationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetIntegrationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetIntegrationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetIntegrationRequestFilterSensitiveLog,
            outputFilterSensitiveLog: IntegrationFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetIntegrationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetIntegrationCommand(input, context);
    };
    GetIntegrationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetIntegrationCommand(output, context);
    };
    return GetIntegrationCommand;
}($Command));
export { GetIntegrationCommand };
