// @aws-sdk/client-api-gateway/dist-es/commands/GetIntegrationResponseCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetIntegrationResponseRequestFilterSensitiveLog, IntegrationResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetIntegrationResponseCommand, serializeAws_restJson1GetIntegrationResponseCommand, } from "../protocols/Aws_restJson1";
var GetIntegrationResponseCommand = (function (_super) {
    __extends(GetIntegrationResponseCommand, _super);
    function GetIntegrationResponseCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetIntegrationResponseCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetIntegrationResponseCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetIntegrationResponseRequestFilterSensitiveLog,
            outputFilterSensitiveLog: IntegrationResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetIntegrationResponseCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetIntegrationResponseCommand(input, context);
    };
    GetIntegrationResponseCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetIntegrationResponseCommand(output, context);
    };
    return GetIntegrationResponseCommand;
}($Command));
export { GetIntegrationResponseCommand };
