// @aws-sdk/client-api-gateway/dist-es/commands/GetModelCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetModelRequestFilterSensitiveLog, ModelFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1GetModelCommand, serializeAws_restJson1GetModelCommand, } from "../protocols/Aws_restJson1";
var GetModelCommand = (function (_super) {
    __extends(GetModelCommand, _super);
    function GetModelCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetModelCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetModelCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetModelRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ModelFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetModelCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetModelCommand(input, context);
    };
    GetModelCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetModelCommand(output, context);
    };
    return GetModelCommand;
}($Command));
export { GetModelCommand };
