// @aws-sdk/client-api-gateway/dist-es/commands/GetModelsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetModelsRequestFilterSensitiveLog, ModelsFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetModelsCommand, serializeAws_restJson1GetModelsCommand, } from "../protocols/Aws_restJson1";
var GetModelsCommand = (function (_super) {
    __extends(GetModelsCommand, _super);
    function GetModelsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetModelsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetModelsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetModelsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ModelsFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetModelsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetModelsCommand(input, context);
    };
    GetModelsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetModelsCommand(output, context);
    };
    return GetModelsCommand;
}($Command));
export { GetModelsCommand };
