// @aws-sdk/client-api-gateway/dist-es/commands/GetRequestValidatorsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetRequestValidatorsRequestFilterSensitiveLog, RequestValidatorsFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetRequestValidatorsCommand, serializeAws_restJson1GetRequestValidatorsCommand, } from "../protocols/Aws_restJson1";
var GetRequestValidatorsCommand = (function (_super) {
    __extends(GetRequestValidatorsCommand, _super);
    function GetRequestValidatorsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetRequestValidatorsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetRequestValidatorsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetRequestValidatorsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: RequestValidatorsFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetRequestValidatorsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetRequestValidatorsCommand(input, context);
    };
    GetRequestValidatorsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetRequestValidatorsCommand(output, context);
    };
    return GetRequestValidatorsCommand;
}($Command));
export { GetRequestValidatorsCommand };
