// @aws-sdk/client-api-gateway/dist-es/commands/GetResourcesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetResourcesRequestFilterSensitiveLog, ResourcesFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetResourcesCommand, serializeAws_restJson1GetResourcesCommand, } from "../protocols/Aws_restJson1";
var GetResourcesCommand = (function (_super) {
    __extends(GetResourcesCommand, _super);
    function GetResourcesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetResourcesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetResourcesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetResourcesRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ResourcesFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetResourcesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetResourcesCommand(input, context);
    };
    GetResourcesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetResourcesCommand(output, context);
    };
    return GetResourcesCommand;
}($Command));
export { GetResourcesCommand };
