// @aws-sdk/client-api-gateway/dist-es/commands/GetSdkTypeCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetSdkTypeRequestFilterSensitiveLog, SdkTypeFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetSdkTypeCommand, serializeAws_restJson1GetSdkTypeCommand, } from "../protocols/Aws_restJson1";
var GetSdkTypeCommand = (function (_super) {
    __extends(GetSdkTypeCommand, _super);
    function GetSdkTypeCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetSdkTypeCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetSdkTypeCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetSdkTypeRequestFilterSensitiveLog,
            outputFilterSensitiveLog: SdkTypeFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetSdkTypeCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetSdkTypeCommand(input, context);
    };
    GetSdkTypeCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetSdkTypeCommand(output, context);
    };
    return GetSdkTypeCommand;
}($Command));
export { GetSdkTypeCommand };
