// @aws-sdk/client-api-gateway/dist-es/commands/GetTagsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetTagsRequestFilterSensitiveLog, TagsFilterSensitiveLog } from "../models/models_0";
import { deserializeAws_restJson1GetTagsCommand, serializeAws_restJson1GetTagsCommand, } from "../protocols/Aws_restJson1";
var GetTagsCommand = (function (_super) {
    __extends(GetTagsCommand, _super);
    function GetTagsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetTagsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetTagsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetTagsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: TagsFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetTagsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetTagsCommand(input, context);
    };
    GetTagsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetTagsCommand(output, context);
    };
    return GetTagsCommand;
}($Command));
export { GetTagsCommand };
