// @aws-sdk/client-api-gateway/dist-es/commands/GetUsagePlanKeysCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetUsagePlanKeysRequestFilterSensitiveLog, UsagePlanKeysFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetUsagePlanKeysCommand, serializeAws_restJson1GetUsagePlanKeysCommand, } from "../protocols/Aws_restJson1";
var GetUsagePlanKeysCommand = (function (_super) {
    __extends(GetUsagePlanKeysCommand, _super);
    function GetUsagePlanKeysCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetUsagePlanKeysCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetUsagePlanKeysCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetUsagePlanKeysRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UsagePlanKeysFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetUsagePlanKeysCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetUsagePlanKeysCommand(input, context);
    };
    GetUsagePlanKeysCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetUsagePlanKeysCommand(output, context);
    };
    return GetUsagePlanKeysCommand;
}($Command));
export { GetUsagePlanKeysCommand };
