// @aws-sdk/client-api-gateway/dist-es/commands/GetUsagePlansCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetUsagePlansRequestFilterSensitiveLog, UsagePlansFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetUsagePlansCommand, serializeAws_restJson1GetUsagePlansCommand, } from "../protocols/Aws_restJson1";
var GetUsagePlansCommand = (function (_super) {
    __extends(GetUsagePlansCommand, _super);
    function GetUsagePlansCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetUsagePlansCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetUsagePlansCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetUsagePlansRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UsagePlansFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetUsagePlansCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetUsagePlansCommand(input, context);
    };
    GetUsagePlansCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetUsagePlansCommand(output, context);
    };
    return GetUsagePlansCommand;
}($Command));
export { GetUsagePlansCommand };
