// @aws-sdk/client-api-gateway/dist-es/commands/GetVpcLinksCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetVpcLinksRequestFilterSensitiveLog, VpcLinksFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1GetVpcLinksCommand, serializeAws_restJson1GetVpcLinksCommand, } from "../protocols/Aws_restJson1";
var GetVpcLinksCommand = (function (_super) {
    __extends(GetVpcLinksCommand, _super);
    function GetVpcLinksCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetVpcLinksCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "GetVpcLinksCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetVpcLinksRequestFilterSensitiveLog,
            outputFilterSensitiveLog: VpcLinksFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetVpcLinksCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetVpcLinksCommand(input, context);
    };
    GetVpcLinksCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetVpcLinksCommand(output, context);
    };
    return GetVpcLinksCommand;
}($Command));
export { GetVpcLinksCommand };
