// @aws-sdk/client-api-gateway/dist-es/commands/ImportDocumentationPartsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DocumentationPartIdsFilterSensitiveLog, ImportDocumentationPartsRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1ImportDocumentationPartsCommand, serializeAws_restJson1ImportDocumentationPartsCommand, } from "../protocols/Aws_restJson1";
var ImportDocumentationPartsCommand = (function (_super) {
    __extends(ImportDocumentationPartsCommand, _super);
    function ImportDocumentationPartsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ImportDocumentationPartsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "ImportDocumentationPartsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ImportDocumentationPartsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: DocumentationPartIdsFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ImportDocumentationPartsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ImportDocumentationPartsCommand(input, context);
    };
    ImportDocumentationPartsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ImportDocumentationPartsCommand(output, context);
    };
    return ImportDocumentationPartsCommand;
}($Command));
export { ImportDocumentationPartsCommand };
