// @aws-sdk/client-api-gateway/dist-es/commands/PutIntegrationResponseCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { IntegrationResponseFilterSensitiveLog, PutIntegrationResponseRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1PutIntegrationResponseCommand, serializeAws_restJson1PutIntegrationResponseCommand, } from "../protocols/Aws_restJson1";
var PutIntegrationResponseCommand = (function (_super) {
    __extends(PutIntegrationResponseCommand, _super);
    function PutIntegrationResponseCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutIntegrationResponseCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "PutIntegrationResponseCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutIntegrationResponseRequestFilterSensitiveLog,
            outputFilterSensitiveLog: IntegrationResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutIntegrationResponseCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1PutIntegrationResponseCommand(input, context);
    };
    PutIntegrationResponseCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1PutIntegrationResponseCommand(output, context);
    };
    return PutIntegrationResponseCommand;
}($Command));
export { PutIntegrationResponseCommand };
