// @aws-sdk/client-api-gateway/dist-es/commands/PutIntegrationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { IntegrationFilterSensitiveLog, PutIntegrationRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1PutIntegrationCommand, serializeAws_restJson1PutIntegrationCommand, } from "../protocols/Aws_restJson1";
var PutIntegrationCommand = (function (_super) {
    __extends(PutIntegrationCommand, _super);
    function PutIntegrationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    PutIntegrationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "PutIntegrationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: PutIntegrationRequestFilterSensitiveLog,
            outputFilterSensitiveLog: IntegrationFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    PutIntegrationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1PutIntegrationCommand(input, context);
    };
    PutIntegrationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1PutIntegrationCommand(output, context);
    };
    return PutIntegrationCommand;
}($Command));
export { PutIntegrationCommand };
