// @aws-sdk/client-api-gateway/dist-es/commands/UpdateModelCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ModelFilterSensitiveLog, UpdateModelRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateModelCommand, serializeAws_restJson1UpdateModelCommand, } from "../protocols/Aws_restJson1";
var UpdateModelCommand = (function (_super) {
    __extends(UpdateModelCommand, _super);
    function UpdateModelCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateModelCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "UpdateModelCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateModelRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ModelFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateModelCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateModelCommand(input, context);
    };
    UpdateModelCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateModelCommand(output, context);
    };
    return UpdateModelCommand;
}($Command));
export { UpdateModelCommand };
