// @aws-sdk/client-api-gateway/dist-es/commands/UpdateClientCertificateCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ClientCertificateFilterSensitiveLog, UpdateClientCertificateRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateClientCertificateCommand, serializeAws_restJson1UpdateClientCertificateCommand, } from "../protocols/Aws_restJson1";
var UpdateClientCertificateCommand = (function (_super) {
    __extends(UpdateClientCertificateCommand, _super);
    function UpdateClientCertificateCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateClientCertificateCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "UpdateClientCertificateCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateClientCertificateRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ClientCertificateFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateClientCertificateCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateClientCertificateCommand(input, context);
    };
    UpdateClientCertificateCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateClientCertificateCommand(output, context);
    };
    return UpdateClientCertificateCommand;
}($Command));
export { UpdateClientCertificateCommand };
