// @aws-sdk/client-api-gateway/dist-es/commands/UpdateUsagePlanCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateUsagePlanRequestFilterSensitiveLog, UsagePlanFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateUsagePlanCommand, serializeAws_restJson1UpdateUsagePlanCommand, } from "../protocols/Aws_restJson1";
var UpdateUsagePlanCommand = (function (_super) {
    __extends(UpdateUsagePlanCommand, _super);
    function UpdateUsagePlanCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateUsagePlanCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "UpdateUsagePlanCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateUsagePlanRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UsagePlanFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateUsagePlanCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateUsagePlanCommand(input, context);
    };
    UpdateUsagePlanCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateUsagePlanCommand(output, context);
    };
    return UpdateUsagePlanCommand;
}($Command));
export { UpdateUsagePlanCommand };
