// @aws-sdk/client-api-gateway/dist-es/commands/UpdateMethodResponseCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { MethodResponseFilterSensitiveLog, UpdateMethodResponseRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateMethodResponseCommand, serializeAws_restJson1UpdateMethodResponseCommand, } from "../protocols/Aws_restJson1";
var UpdateMethodResponseCommand = (function (_super) {
    __extends(UpdateMethodResponseCommand, _super);
    function UpdateMethodResponseCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateMethodResponseCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "UpdateMethodResponseCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateMethodResponseRequestFilterSensitiveLog,
            outputFilterSensitiveLog: MethodResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateMethodResponseCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateMethodResponseCommand(input, context);
    };
    UpdateMethodResponseCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateMethodResponseCommand(output, context);
    };
    return UpdateMethodResponseCommand;
}($Command));
export { UpdateMethodResponseCommand };
