// @aws-sdk/client-api-gateway/dist-es/commands/UpdateVpcLinkCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateVpcLinkRequestFilterSensitiveLog, VpcLinkFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateVpcLinkCommand, serializeAws_restJson1UpdateVpcLinkCommand, } from "../protocols/Aws_restJson1";
var UpdateVpcLinkCommand = (function (_super) {
    __extends(UpdateVpcLinkCommand, _super);
    function UpdateVpcLinkCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateVpcLinkCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "UpdateVpcLinkCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateVpcLinkRequestFilterSensitiveLog,
            outputFilterSensitiveLog: VpcLinkFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateVpcLinkCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateVpcLinkCommand(input, context);
    };
    UpdateVpcLinkCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateVpcLinkCommand(output, context);
    };
    return UpdateVpcLinkCommand;
}($Command));
export { UpdateVpcLinkCommand };
