// @aws-sdk/client-api-gateway/dist-es/commands/UpdateUsageCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateUsageRequestFilterSensitiveLog, UsageFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateUsageCommand, serializeAws_restJson1UpdateUsageCommand, } from "../protocols/Aws_restJson1";
var UpdateUsageCommand = (function (_super) {
    __extends(UpdateUsageCommand, _super);
    function UpdateUsageCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateUsageCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "UpdateUsageCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateUsageRequestFilterSensitiveLog,
            outputFilterSensitiveLog: UsageFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateUsageCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateUsageCommand(input, context);
    };
    UpdateUsageCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateUsageCommand(output, context);
    };
    return UpdateUsageCommand;
}($Command));
export { UpdateUsageCommand };
