// @aws-sdk/client-api-gateway/dist-es/commands/UpdateApiKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ApiKeyFilterSensitiveLog, UpdateApiKeyRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateApiKeyCommand, serializeAws_restJson1UpdateApiKeyCommand, } from "../protocols/Aws_restJson1";
var UpdateApiKeyCommand = (function (_super) {
    __extends(UpdateApiKeyCommand, _super);
    function UpdateApiKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateApiKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "UpdateApiKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateApiKeyRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ApiKeyFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateApiKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateApiKeyCommand(input, context);
    };
    UpdateApiKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateApiKeyCommand(output, context);
    };
    return UpdateApiKeyCommand;
}($Command));
export { UpdateApiKeyCommand };
