// @aws-sdk/client-api-gateway/dist-es/commands/UpdateAuthorizerCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AuthorizerFilterSensitiveLog, UpdateAuthorizerRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateAuthorizerCommand, serializeAws_restJson1UpdateAuthorizerCommand, } from "../protocols/Aws_restJson1";
var UpdateAuthorizerCommand = (function (_super) {
    __extends(UpdateAuthorizerCommand, _super);
    function UpdateAuthorizerCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateAuthorizerCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "UpdateAuthorizerCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateAuthorizerRequestFilterSensitiveLog,
            outputFilterSensitiveLog: AuthorizerFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateAuthorizerCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateAuthorizerCommand(input, context);
    };
    UpdateAuthorizerCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateAuthorizerCommand(output, context);
    };
    return UpdateAuthorizerCommand;
}($Command));
export { UpdateAuthorizerCommand };
