// @aws-sdk/client-api-gateway/dist-es/commands/UpdateResourceCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ResourceFilterSensitiveLog, UpdateResourceRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateResourceCommand, serializeAws_restJson1UpdateResourceCommand, } from "../protocols/Aws_restJson1";
var UpdateResourceCommand = (function (_super) {
    __extends(UpdateResourceCommand, _super);
    function UpdateResourceCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateResourceCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "UpdateResourceCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateResourceRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ResourceFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateResourceCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateResourceCommand(input, context);
    };
    UpdateResourceCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateResourceCommand(output, context);
    };
    return UpdateResourceCommand;
}($Command));
export { UpdateResourceCommand };
