// @aws-sdk/client-api-gateway/dist-es/commands/UpdateIntegrationCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { IntegrationFilterSensitiveLog, UpdateIntegrationRequestFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_restJson1UpdateIntegrationCommand, serializeAws_restJson1UpdateIntegrationCommand, } from "../protocols/Aws_restJson1";
var UpdateIntegrationCommand = (function (_super) {
    __extends(UpdateIntegrationCommand, _super);
    function UpdateIntegrationCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateIntegrationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "APIGatewayClient";
        var commandName = "UpdateIntegrationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateIntegrationRequestFilterSensitiveLog,
            outputFilterSensitiveLog: IntegrationFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateIntegrationCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdateIntegrationCommand(input, context);
    };
    UpdateIntegrationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdateIntegrationCommand(output, context);
    };
    return UpdateIntegrationCommand;
}($Command));
export { UpdateIntegrationCommand };
