// @aws-sdk/client-api-gateway/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MutualTlsAuthenticationFilterSensitiveLog = exports.CreateDomainNameRequestFilterSensitiveLog = exports.MutualTlsAuthenticationInputFilterSensitiveLog = exports.EndpointConfigurationFilterSensitiveLog = exports.DocumentationVersionFilterSensitiveLog = exports.CreateDocumentationVersionRequestFilterSensitiveLog = exports.DocumentationPartFilterSensitiveLog = exports.CreateDocumentationPartRequestFilterSensitiveLog = exports.DocumentationPartLocationFilterSensitiveLog = exports.DeploymentFilterSensitiveLog = exports.MethodSnapshotFilterSensitiveLog = exports.CreateDeploymentRequestFilterSensitiveLog = exports.DeploymentCanarySettingsFilterSensitiveLog = exports.CreateBasePathMappingRequestFilterSensitiveLog = exports.BasePathMappingFilterSensitiveLog = exports.CreateAuthorizerRequestFilterSensitiveLog = exports.CreateApiKeyRequestFilterSensitiveLog = exports.StageKeyFilterSensitiveLog = exports.AuthorizersFilterSensitiveLog = exports.AuthorizerFilterSensitiveLog = exports.ApiStageFilterSensitiveLog = exports.ApiKeysFilterSensitiveLog = exports.ApiKeyIdsFilterSensitiveLog = exports.ApiKeyFilterSensitiveLog = exports.AccountFilterSensitiveLog = exports.ThrottleSettingsFilterSensitiveLog = exports.AccessLogSettingsFilterSensitiveLog = exports.PutMode = exports.LocationStatusType = exports.GatewayResponseType = exports.VpcLinkStatus = exports.QuotaPeriodType = exports.UnauthorizedCacheControlHeaderStrategy = exports.CacheClusterStatus = exports.IntegrationType = exports.ContentHandlingStrategy = exports.ConnectionType = exports.DomainNameStatus = exports.DocumentationPartType = exports.ServiceUnavailableException = exports.CacheClusterSize = exports.UnauthorizedException = exports.TooManyRequestsException = exports.NotFoundException = exports.LimitExceededException = exports.ConflictException = exports.BadRequestException = exports.AuthorizerType = exports.ApiKeySourceType = exports.ApiKeysFormat = void 0;
exports.GenerateClientCertificateRequestFilterSensitiveLog = exports.ClientCertificateFilterSensitiveLog = exports.FlushStageCacheRequestFilterSensitiveLog = exports.FlushStageAuthorizersCacheRequestFilterSensitiveLog = exports.DeleteVpcLinkRequestFilterSensitiveLog = exports.DeleteUsagePlanKeyRequestFilterSensitiveLog = exports.DeleteUsagePlanRequestFilterSensitiveLog = exports.DeleteStageRequestFilterSensitiveLog = exports.DeleteRestApiRequestFilterSensitiveLog = exports.DeleteResourceRequestFilterSensitiveLog = exports.DeleteRequestValidatorRequestFilterSensitiveLog = exports.DeleteModelRequestFilterSensitiveLog = exports.DeleteMethodResponseRequestFilterSensitiveLog = exports.DeleteMethodRequestFilterSensitiveLog = exports.DeleteIntegrationResponseRequestFilterSensitiveLog = exports.DeleteIntegrationRequestFilterSensitiveLog = exports.DeleteGatewayResponseRequestFilterSensitiveLog = exports.DeleteDomainNameRequestFilterSensitiveLog = exports.DeleteDocumentationVersionRequestFilterSensitiveLog = exports.DeleteDocumentationPartRequestFilterSensitiveLog = exports.DeleteDeploymentRequestFilterSensitiveLog = exports.DeleteClientCertificateRequestFilterSensitiveLog = exports.DeleteBasePathMappingRequestFilterSensitiveLog = exports.DeleteAuthorizerRequestFilterSensitiveLog = exports.DeleteApiKeyRequestFilterSensitiveLog = exports.VpcLinkFilterSensitiveLog = exports.CreateVpcLinkRequestFilterSensitiveLog = exports.UsagePlanKeyFilterSensitiveLog = exports.CreateUsagePlanKeyRequestFilterSensitiveLog = exports.UsagePlanFilterSensitiveLog = exports.CreateUsagePlanRequestFilterSensitiveLog = exports.QuotaSettingsFilterSensitiveLog = exports.StageFilterSensitiveLog = exports.MethodSettingFilterSensitiveLog = exports.CreateStageRequestFilterSensitiveLog = exports.CanarySettingsFilterSensitiveLog = exports.RestApiFilterSensitiveLog = exports.CreateRestApiRequestFilterSensitiveLog = exports.ResourceFilterSensitiveLog = exports.MethodFilterSensitiveLog = exports.MethodResponseFilterSensitiveLog = exports.IntegrationFilterSensitiveLog = exports.TlsConfigFilterSensitiveLog = exports.IntegrationResponseFilterSensitiveLog = exports.CreateResourceRequestFilterSensitiveLog = exports.RequestValidatorFilterSensitiveLog = exports.CreateRequestValidatorRequestFilterSensitiveLog = exports.ModelFilterSensitiveLog = exports.CreateModelRequestFilterSensitiveLog = exports.DomainNameFilterSensitiveLog = void 0;
exports.GetSdkTypeRequestFilterSensitiveLog = exports.SdkResponseFilterSensitiveLog = exports.GetSdkRequestFilterSensitiveLog = exports.RestApisFilterSensitiveLog = exports.GetRestApisRequestFilterSensitiveLog = exports.GetRestApiRequestFilterSensitiveLog = exports.ResourcesFilterSensitiveLog = exports.GetResourcesRequestFilterSensitiveLog = exports.GetResourceRequestFilterSensitiveLog = exports.RequestValidatorsFilterSensitiveLog = exports.GetRequestValidatorsRequestFilterSensitiveLog = exports.GetRequestValidatorRequestFilterSensitiveLog = exports.TemplateFilterSensitiveLog = exports.GetModelTemplateRequestFilterSensitiveLog = exports.ModelsFilterSensitiveLog = exports.GetModelsRequestFilterSensitiveLog = exports.GetModelRequestFilterSensitiveLog = exports.GetMethodResponseRequestFilterSensitiveLog = exports.GetMethodRequestFilterSensitiveLog = exports.GetIntegrationResponseRequestFilterSensitiveLog = exports.GetIntegrationRequestFilterSensitiveLog = exports.GetGatewayResponsesRequestFilterSensitiveLog = exports.GatewayResponsesFilterSensitiveLog = exports.GetGatewayResponseRequestFilterSensitiveLog = exports.GatewayResponseFilterSensitiveLog = exports.GetExportRequestFilterSensitiveLog = exports.ExportResponseFilterSensitiveLog = exports.GetDomainNamesRequestFilterSensitiveLog = exports.DomainNamesFilterSensitiveLog = exports.GetDomainNameRequestFilterSensitiveLog = exports.GetDocumentationVersionsRequestFilterSensitiveLog = exports.DocumentationVersionsFilterSensitiveLog = exports.GetDocumentationVersionRequestFilterSensitiveLog = exports.GetDocumentationPartsRequestFilterSensitiveLog = exports.DocumentationPartsFilterSensitiveLog = exports.GetDocumentationPartRequestFilterSensitiveLog = exports.GetDeploymentsRequestFilterSensitiveLog = exports.DeploymentsFilterSensitiveLog = exports.GetDeploymentRequestFilterSensitiveLog = exports.GetClientCertificatesRequestFilterSensitiveLog = exports.ClientCertificatesFilterSensitiveLog = exports.GetClientCertificateRequestFilterSensitiveLog = exports.GetBasePathMappingsRequestFilterSensitiveLog = exports.BasePathMappingsFilterSensitiveLog = exports.GetBasePathMappingRequestFilterSensitiveLog = exports.GetAuthorizersRequestFilterSensitiveLog = exports.GetAuthorizerRequestFilterSensitiveLog = exports.GetApiKeysRequestFilterSensitiveLog = exports.GetApiKeyRequestFilterSensitiveLog = exports.GetAccountRequestFilterSensitiveLog = void 0;
exports.UpdateMethodRequestFilterSensitiveLog = exports.UpdateIntegrationResponseRequestFilterSensitiveLog = exports.UpdateIntegrationRequestFilterSensitiveLog = exports.UpdateGatewayResponseRequestFilterSensitiveLog = exports.UpdateDomainNameRequestFilterSensitiveLog = exports.UpdateDocumentationVersionRequestFilterSensitiveLog = exports.UpdateDocumentationPartRequestFilterSensitiveLog = exports.UpdateDeploymentRequestFilterSensitiveLog = exports.UpdateClientCertificateRequestFilterSensitiveLog = exports.UpdateBasePathMappingRequestFilterSensitiveLog = exports.UpdateAuthorizerRequestFilterSensitiveLog = exports.UpdateApiKeyRequestFilterSensitiveLog = exports.UpdateAccountRequestFilterSensitiveLog = exports.PatchOperationFilterSensitiveLog = exports.UntagResourceRequestFilterSensitiveLog = exports.TestInvokeMethodResponseFilterSensitiveLog = exports.TestInvokeMethodRequestFilterSensitiveLog = exports.TestInvokeAuthorizerResponseFilterSensitiveLog = exports.TestInvokeAuthorizerRequestFilterSensitiveLog = exports.TagResourceRequestFilterSensitiveLog = exports.PutRestApiRequestFilterSensitiveLog = exports.PutMethodResponseRequestFilterSensitiveLog = exports.PutMethodRequestFilterSensitiveLog = exports.PutIntegrationResponseRequestFilterSensitiveLog = exports.PutIntegrationRequestFilterSensitiveLog = exports.PutGatewayResponseRequestFilterSensitiveLog = exports.ImportRestApiRequestFilterSensitiveLog = exports.ImportDocumentationPartsRequestFilterSensitiveLog = exports.DocumentationPartIdsFilterSensitiveLog = exports.ImportApiKeysRequestFilterSensitiveLog = exports.VpcLinksFilterSensitiveLog = exports.GetVpcLinksRequestFilterSensitiveLog = exports.GetVpcLinkRequestFilterSensitiveLog = exports.UsagePlansFilterSensitiveLog = exports.GetUsagePlansRequestFilterSensitiveLog = exports.UsagePlanKeysFilterSensitiveLog = exports.GetUsagePlanKeysRequestFilterSensitiveLog = exports.GetUsagePlanKeyRequestFilterSensitiveLog = exports.GetUsagePlanRequestFilterSensitiveLog = exports.UsageFilterSensitiveLog = exports.GetUsageRequestFilterSensitiveLog = exports.TagsFilterSensitiveLog = exports.GetTagsRequestFilterSensitiveLog = exports.StagesFilterSensitiveLog = exports.GetStagesRequestFilterSensitiveLog = exports.GetStageRequestFilterSensitiveLog = exports.SdkTypesFilterSensitiveLog = exports.GetSdkTypesRequestFilterSensitiveLog = exports.SdkTypeFilterSensitiveLog = exports.SdkConfigurationPropertyFilterSensitiveLog = void 0;
exports.UpdateVpcLinkRequestFilterSensitiveLog = exports.UpdateUsagePlanRequestFilterSensitiveLog = exports.UpdateUsageRequestFilterSensitiveLog = exports.UpdateStageRequestFilterSensitiveLog = exports.UpdateRestApiRequestFilterSensitiveLog = exports.UpdateResourceRequestFilterSensitiveLog = exports.UpdateRequestValidatorRequestFilterSensitiveLog = exports.UpdateModelRequestFilterSensitiveLog = exports.UpdateMethodResponseRequestFilterSensitiveLog = void 0;
const APIGatewayServiceException_1 = require("./APIGatewayServiceException");
var ApiKeysFormat;
(function (ApiKeysFormat) {
    ApiKeysFormat["csv"] = "csv";
})(ApiKeysFormat = exports.ApiKeysFormat || (exports.ApiKeysFormat = {}));
var ApiKeySourceType;
(function (ApiKeySourceType) {
    ApiKeySourceType["AUTHORIZER"] = "AUTHORIZER";
    ApiKeySourceType["HEADER"] = "HEADER";
})(ApiKeySourceType = exports.ApiKeySourceType || (exports.ApiKeySourceType = {}));
var AuthorizerType;
(function (AuthorizerType) {
    AuthorizerType["COGNITO_USER_POOLS"] = "COGNITO_USER_POOLS";
    AuthorizerType["REQUEST"] = "REQUEST";
    AuthorizerType["TOKEN"] = "TOKEN";
})(AuthorizerType = exports.AuthorizerType || (exports.AuthorizerType = {}));
class BadRequestException extends APIGatewayServiceException_1.APIGatewayServiceException {
    constructor(opts) {
        super({
            name: "BadRequestException",
            $fault: "client",
            ...opts,
        });
        this.name = "BadRequestException";
        this.$fault = "client";
        Object.setPrototypeOf(this, BadRequestException.prototype);
    }
}
exports.BadRequestException = BadRequestException;
class ConflictException extends APIGatewayServiceException_1.APIGatewayServiceException {
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
exports.ConflictException = ConflictException;
class LimitExceededException extends APIGatewayServiceException_1.APIGatewayServiceException {
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "LimitExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, LimitExceededException.prototype);
        this.retryAfterSeconds = opts.retryAfterSeconds;
    }
}
exports.LimitExceededException = LimitExceededException;
class NotFoundException extends APIGatewayServiceException_1.APIGatewayServiceException {
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "NotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, NotFoundException.prototype);
    }
}
exports.NotFoundException = NotFoundException;
class TooManyRequestsException extends APIGatewayServiceException_1.APIGatewayServiceException {
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyRequestsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.retryAfterSeconds = opts.retryAfterSeconds;
    }
}
exports.TooManyRequestsException = TooManyRequestsException;
class UnauthorizedException extends APIGatewayServiceException_1.APIGatewayServiceException {
    constructor(opts) {
        super({
            name: "UnauthorizedException",
            $fault: "client",
            ...opts,
        });
        this.name = "UnauthorizedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, UnauthorizedException.prototype);
    }
}
exports.UnauthorizedException = UnauthorizedException;
var CacheClusterSize;
(function (CacheClusterSize) {
    CacheClusterSize["SIZE_0_POINT_5_GB"] = "0.5";
    CacheClusterSize["SIZE_118_GB"] = "118";
    CacheClusterSize["SIZE_13_POINT_5_GB"] = "13.5";
    CacheClusterSize["SIZE_1_POINT_6_GB"] = "1.6";
    CacheClusterSize["SIZE_237_GB"] = "237";
    CacheClusterSize["SIZE_28_POINT_4_GB"] = "28.4";
    CacheClusterSize["SIZE_58_POINT_2_GB"] = "58.2";
    CacheClusterSize["SIZE_6_POINT_1_GB"] = "6.1";
})(CacheClusterSize = exports.CacheClusterSize || (exports.CacheClusterSize = {}));
class ServiceUnavailableException extends APIGatewayServiceException_1.APIGatewayServiceException {
    constructor(opts) {
        super({
            name: "ServiceUnavailableException",
            $fault: "server",
            ...opts,
        });
        this.name = "ServiceUnavailableException";
        this.$fault = "server";
        Object.setPrototypeOf(this, ServiceUnavailableException.prototype);
        this.retryAfterSeconds = opts.retryAfterSeconds;
    }
}
exports.ServiceUnavailableException = ServiceUnavailableException;
var DocumentationPartType;
(function (DocumentationPartType) {
    DocumentationPartType["API"] = "API";
    DocumentationPartType["AUTHORIZER"] = "AUTHORIZER";
    DocumentationPartType["METHOD"] = "METHOD";
    DocumentationPartType["MODEL"] = "MODEL";
    DocumentationPartType["PATH_PARAMETER"] = "PATH_PARAMETER";
    DocumentationPartType["QUERY_PARAMETER"] = "QUERY_PARAMETER";
    DocumentationPartType["REQUEST_BODY"] = "REQUEST_BODY";
    DocumentationPartType["REQUEST_HEADER"] = "REQUEST_HEADER";
    DocumentationPartType["RESOURCE"] = "RESOURCE";
    DocumentationPartType["RESPONSE"] = "RESPONSE";
    DocumentationPartType["RESPONSE_BODY"] = "RESPONSE_BODY";
    DocumentationPartType["RESPONSE_HEADER"] = "RESPONSE_HEADER";
})(DocumentationPartType = exports.DocumentationPartType || (exports.DocumentationPartType = {}));
var DomainNameStatus;
(function (DomainNameStatus) {
    DomainNameStatus["AVAILABLE"] = "AVAILABLE";
    DomainNameStatus["PENDING"] = "PENDING";
    DomainNameStatus["PENDING_CERTIFICATE_REIMPORT"] = "PENDING_CERTIFICATE_REIMPORT";
    DomainNameStatus["PENDING_OWNERSHIP_VERIFICATION"] = "PENDING_OWNERSHIP_VERIFICATION";
    DomainNameStatus["UPDATING"] = "UPDATING";
})(DomainNameStatus = exports.DomainNameStatus || (exports.DomainNameStatus = {}));
var ConnectionType;
(function (ConnectionType) {
    ConnectionType["INTERNET"] = "INTERNET";
    ConnectionType["VPC_LINK"] = "VPC_LINK";
})(ConnectionType = exports.ConnectionType || (exports.ConnectionType = {}));
var ContentHandlingStrategy;
(function (ContentHandlingStrategy) {
    ContentHandlingStrategy["CONVERT_TO_BINARY"] = "CONVERT_TO_BINARY";
    ContentHandlingStrategy["CONVERT_TO_TEXT"] = "CONVERT_TO_TEXT";
})(ContentHandlingStrategy = exports.ContentHandlingStrategy || (exports.ContentHandlingStrategy = {}));
var IntegrationType;
(function (IntegrationType) {
    IntegrationType["AWS"] = "AWS";
    IntegrationType["AWS_PROXY"] = "AWS_PROXY";
    IntegrationType["HTTP"] = "HTTP";
    IntegrationType["HTTP_PROXY"] = "HTTP_PROXY";
    IntegrationType["MOCK"] = "MOCK";
})(IntegrationType = exports.IntegrationType || (exports.IntegrationType = {}));
var CacheClusterStatus;
(function (CacheClusterStatus) {
    CacheClusterStatus["AVAILABLE"] = "AVAILABLE";
    CacheClusterStatus["CREATE_IN_PROGRESS"] = "CREATE_IN_PROGRESS";
    CacheClusterStatus["DELETE_IN_PROGRESS"] = "DELETE_IN_PROGRESS";
    CacheClusterStatus["FLUSH_IN_PROGRESS"] = "FLUSH_IN_PROGRESS";
    CacheClusterStatus["NOT_AVAILABLE"] = "NOT_AVAILABLE";
})(CacheClusterStatus = exports.CacheClusterStatus || (exports.CacheClusterStatus = {}));
var UnauthorizedCacheControlHeaderStrategy;
(function (UnauthorizedCacheControlHeaderStrategy) {
    UnauthorizedCacheControlHeaderStrategy["FAIL_WITH_403"] = "FAIL_WITH_403";
    UnauthorizedCacheControlHeaderStrategy["SUCCEED_WITHOUT_RESPONSE_HEADER"] = "SUCCEED_WITHOUT_RESPONSE_HEADER";
    UnauthorizedCacheControlHeaderStrategy["SUCCEED_WITH_RESPONSE_HEADER"] = "SUCCEED_WITH_RESPONSE_HEADER";
})(UnauthorizedCacheControlHeaderStrategy = exports.UnauthorizedCacheControlHeaderStrategy || (exports.UnauthorizedCacheControlHeaderStrategy = {}));
var QuotaPeriodType;
(function (QuotaPeriodType) {
    QuotaPeriodType["DAY"] = "DAY";
    QuotaPeriodType["MONTH"] = "MONTH";
    QuotaPeriodType["WEEK"] = "WEEK";
})(QuotaPeriodType = exports.QuotaPeriodType || (exports.QuotaPeriodType = {}));
var VpcLinkStatus;
(function (VpcLinkStatus) {
    VpcLinkStatus["AVAILABLE"] = "AVAILABLE";
    VpcLinkStatus["DELETING"] = "DELETING";
    VpcLinkStatus["FAILED"] = "FAILED";
    VpcLinkStatus["PENDING"] = "PENDING";
})(VpcLinkStatus = exports.VpcLinkStatus || (exports.VpcLinkStatus = {}));
var GatewayResponseType;
(function (GatewayResponseType) {
    GatewayResponseType["ACCESS_DENIED"] = "ACCESS_DENIED";
    GatewayResponseType["API_CONFIGURATION_ERROR"] = "API_CONFIGURATION_ERROR";
    GatewayResponseType["AUTHORIZER_CONFIGURATION_ERROR"] = "AUTHORIZER_CONFIGURATION_ERROR";
    GatewayResponseType["AUTHORIZER_FAILURE"] = "AUTHORIZER_FAILURE";
    GatewayResponseType["BAD_REQUEST_BODY"] = "BAD_REQUEST_BODY";
    GatewayResponseType["BAD_REQUEST_PARAMETERS"] = "BAD_REQUEST_PARAMETERS";
    GatewayResponseType["DEFAULT_4XX"] = "DEFAULT_4XX";
    GatewayResponseType["DEFAULT_5XX"] = "DEFAULT_5XX";
    GatewayResponseType["EXPIRED_TOKEN"] = "EXPIRED_TOKEN";
    GatewayResponseType["INTEGRATION_FAILURE"] = "INTEGRATION_FAILURE";
    GatewayResponseType["INTEGRATION_TIMEOUT"] = "INTEGRATION_TIMEOUT";
    GatewayResponseType["INVALID_API_KEY"] = "INVALID_API_KEY";
    GatewayResponseType["INVALID_SIGNATURE"] = "INVALID_SIGNATURE";
    GatewayResponseType["MISSING_AUTHENTICATION_TOKEN"] = "MISSING_AUTHENTICATION_TOKEN";
    GatewayResponseType["QUOTA_EXCEEDED"] = "QUOTA_EXCEEDED";
    GatewayResponseType["REQUEST_TOO_LARGE"] = "REQUEST_TOO_LARGE";
    GatewayResponseType["RESOURCE_NOT_FOUND"] = "RESOURCE_NOT_FOUND";
    GatewayResponseType["THROTTLED"] = "THROTTLED";
    GatewayResponseType["UNAUTHORIZED"] = "UNAUTHORIZED";
    GatewayResponseType["UNSUPPORTED_MEDIA_TYPE"] = "UNSUPPORTED_MEDIA_TYPE";
    GatewayResponseType["WAF_FILTERED"] = "WAF_FILTERED";
})(GatewayResponseType = exports.GatewayResponseType || (exports.GatewayResponseType = {}));
var LocationStatusType;
(function (LocationStatusType) {
    LocationStatusType["DOCUMENTED"] = "DOCUMENTED";
    LocationStatusType["UNDOCUMENTED"] = "UNDOCUMENTED";
})(LocationStatusType = exports.LocationStatusType || (exports.LocationStatusType = {}));
var PutMode;
(function (PutMode) {
    PutMode["Merge"] = "merge";
    PutMode["Overwrite"] = "overwrite";
})(PutMode = exports.PutMode || (exports.PutMode = {}));
const AccessLogSettingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AccessLogSettingsFilterSensitiveLog = AccessLogSettingsFilterSensitiveLog;
const ThrottleSettingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ThrottleSettingsFilterSensitiveLog = ThrottleSettingsFilterSensitiveLog;
const AccountFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AccountFilterSensitiveLog = AccountFilterSensitiveLog;
const ApiKeyFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApiKeyFilterSensitiveLog = ApiKeyFilterSensitiveLog;
const ApiKeyIdsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApiKeyIdsFilterSensitiveLog = ApiKeyIdsFilterSensitiveLog;
const ApiKeysFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApiKeysFilterSensitiveLog = ApiKeysFilterSensitiveLog;
const ApiStageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ApiStageFilterSensitiveLog = ApiStageFilterSensitiveLog;
const AuthorizerFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AuthorizerFilterSensitiveLog = AuthorizerFilterSensitiveLog;
const AuthorizersFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AuthorizersFilterSensitiveLog = AuthorizersFilterSensitiveLog;
const StageKeyFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StageKeyFilterSensitiveLog = StageKeyFilterSensitiveLog;
const CreateApiKeyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateApiKeyRequestFilterSensitiveLog = CreateApiKeyRequestFilterSensitiveLog;
const CreateAuthorizerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateAuthorizerRequestFilterSensitiveLog = CreateAuthorizerRequestFilterSensitiveLog;
const BasePathMappingFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BasePathMappingFilterSensitiveLog = BasePathMappingFilterSensitiveLog;
const CreateBasePathMappingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateBasePathMappingRequestFilterSensitiveLog = CreateBasePathMappingRequestFilterSensitiveLog;
const DeploymentCanarySettingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeploymentCanarySettingsFilterSensitiveLog = DeploymentCanarySettingsFilterSensitiveLog;
const CreateDeploymentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDeploymentRequestFilterSensitiveLog = CreateDeploymentRequestFilterSensitiveLog;
const MethodSnapshotFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MethodSnapshotFilterSensitiveLog = MethodSnapshotFilterSensitiveLog;
const DeploymentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeploymentFilterSensitiveLog = DeploymentFilterSensitiveLog;
const DocumentationPartLocationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DocumentationPartLocationFilterSensitiveLog = DocumentationPartLocationFilterSensitiveLog;
const CreateDocumentationPartRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDocumentationPartRequestFilterSensitiveLog = CreateDocumentationPartRequestFilterSensitiveLog;
const DocumentationPartFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DocumentationPartFilterSensitiveLog = DocumentationPartFilterSensitiveLog;
const CreateDocumentationVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDocumentationVersionRequestFilterSensitiveLog = CreateDocumentationVersionRequestFilterSensitiveLog;
const DocumentationVersionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DocumentationVersionFilterSensitiveLog = DocumentationVersionFilterSensitiveLog;
const EndpointConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EndpointConfigurationFilterSensitiveLog = EndpointConfigurationFilterSensitiveLog;
const MutualTlsAuthenticationInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MutualTlsAuthenticationInputFilterSensitiveLog = MutualTlsAuthenticationInputFilterSensitiveLog;
const CreateDomainNameRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDomainNameRequestFilterSensitiveLog = CreateDomainNameRequestFilterSensitiveLog;
const MutualTlsAuthenticationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MutualTlsAuthenticationFilterSensitiveLog = MutualTlsAuthenticationFilterSensitiveLog;
const DomainNameFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DomainNameFilterSensitiveLog = DomainNameFilterSensitiveLog;
const CreateModelRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateModelRequestFilterSensitiveLog = CreateModelRequestFilterSensitiveLog;
const ModelFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ModelFilterSensitiveLog = ModelFilterSensitiveLog;
const CreateRequestValidatorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateRequestValidatorRequestFilterSensitiveLog = CreateRequestValidatorRequestFilterSensitiveLog;
const RequestValidatorFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RequestValidatorFilterSensitiveLog = RequestValidatorFilterSensitiveLog;
const CreateResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateResourceRequestFilterSensitiveLog = CreateResourceRequestFilterSensitiveLog;
const IntegrationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.IntegrationResponseFilterSensitiveLog = IntegrationResponseFilterSensitiveLog;
const TlsConfigFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TlsConfigFilterSensitiveLog = TlsConfigFilterSensitiveLog;
const IntegrationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.IntegrationFilterSensitiveLog = IntegrationFilterSensitiveLog;
const MethodResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MethodResponseFilterSensitiveLog = MethodResponseFilterSensitiveLog;
const MethodFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MethodFilterSensitiveLog = MethodFilterSensitiveLog;
const ResourceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ResourceFilterSensitiveLog = ResourceFilterSensitiveLog;
const CreateRestApiRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateRestApiRequestFilterSensitiveLog = CreateRestApiRequestFilterSensitiveLog;
const RestApiFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RestApiFilterSensitiveLog = RestApiFilterSensitiveLog;
const CanarySettingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CanarySettingsFilterSensitiveLog = CanarySettingsFilterSensitiveLog;
const CreateStageRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateStageRequestFilterSensitiveLog = CreateStageRequestFilterSensitiveLog;
const MethodSettingFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MethodSettingFilterSensitiveLog = MethodSettingFilterSensitiveLog;
const StageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StageFilterSensitiveLog = StageFilterSensitiveLog;
const QuotaSettingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QuotaSettingsFilterSensitiveLog = QuotaSettingsFilterSensitiveLog;
const CreateUsagePlanRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateUsagePlanRequestFilterSensitiveLog = CreateUsagePlanRequestFilterSensitiveLog;
const UsagePlanFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UsagePlanFilterSensitiveLog = UsagePlanFilterSensitiveLog;
const CreateUsagePlanKeyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateUsagePlanKeyRequestFilterSensitiveLog = CreateUsagePlanKeyRequestFilterSensitiveLog;
const UsagePlanKeyFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UsagePlanKeyFilterSensitiveLog = UsagePlanKeyFilterSensitiveLog;
const CreateVpcLinkRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateVpcLinkRequestFilterSensitiveLog = CreateVpcLinkRequestFilterSensitiveLog;
const VpcLinkFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.VpcLinkFilterSensitiveLog = VpcLinkFilterSensitiveLog;
const DeleteApiKeyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteApiKeyRequestFilterSensitiveLog = DeleteApiKeyRequestFilterSensitiveLog;
const DeleteAuthorizerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteAuthorizerRequestFilterSensitiveLog = DeleteAuthorizerRequestFilterSensitiveLog;
const DeleteBasePathMappingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteBasePathMappingRequestFilterSensitiveLog = DeleteBasePathMappingRequestFilterSensitiveLog;
const DeleteClientCertificateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteClientCertificateRequestFilterSensitiveLog = DeleteClientCertificateRequestFilterSensitiveLog;
const DeleteDeploymentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDeploymentRequestFilterSensitiveLog = DeleteDeploymentRequestFilterSensitiveLog;
const DeleteDocumentationPartRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDocumentationPartRequestFilterSensitiveLog = DeleteDocumentationPartRequestFilterSensitiveLog;
const DeleteDocumentationVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDocumentationVersionRequestFilterSensitiveLog = DeleteDocumentationVersionRequestFilterSensitiveLog;
const DeleteDomainNameRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDomainNameRequestFilterSensitiveLog = DeleteDomainNameRequestFilterSensitiveLog;
const DeleteGatewayResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteGatewayResponseRequestFilterSensitiveLog = DeleteGatewayResponseRequestFilterSensitiveLog;
const DeleteIntegrationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteIntegrationRequestFilterSensitiveLog = DeleteIntegrationRequestFilterSensitiveLog;
const DeleteIntegrationResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteIntegrationResponseRequestFilterSensitiveLog = DeleteIntegrationResponseRequestFilterSensitiveLog;
const DeleteMethodRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteMethodRequestFilterSensitiveLog = DeleteMethodRequestFilterSensitiveLog;
const DeleteMethodResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteMethodResponseRequestFilterSensitiveLog = DeleteMethodResponseRequestFilterSensitiveLog;
const DeleteModelRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteModelRequestFilterSensitiveLog = DeleteModelRequestFilterSensitiveLog;
const DeleteRequestValidatorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteRequestValidatorRequestFilterSensitiveLog = DeleteRequestValidatorRequestFilterSensitiveLog;
const DeleteResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteResourceRequestFilterSensitiveLog = DeleteResourceRequestFilterSensitiveLog;
const DeleteRestApiRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteRestApiRequestFilterSensitiveLog = DeleteRestApiRequestFilterSensitiveLog;
const DeleteStageRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteStageRequestFilterSensitiveLog = DeleteStageRequestFilterSensitiveLog;
const DeleteUsagePlanRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteUsagePlanRequestFilterSensitiveLog = DeleteUsagePlanRequestFilterSensitiveLog;
const DeleteUsagePlanKeyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteUsagePlanKeyRequestFilterSensitiveLog = DeleteUsagePlanKeyRequestFilterSensitiveLog;
const DeleteVpcLinkRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteVpcLinkRequestFilterSensitiveLog = DeleteVpcLinkRequestFilterSensitiveLog;
const FlushStageAuthorizersCacheRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FlushStageAuthorizersCacheRequestFilterSensitiveLog = FlushStageAuthorizersCacheRequestFilterSensitiveLog;
const FlushStageCacheRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FlushStageCacheRequestFilterSensitiveLog = FlushStageCacheRequestFilterSensitiveLog;
const ClientCertificateFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ClientCertificateFilterSensitiveLog = ClientCertificateFilterSensitiveLog;
const GenerateClientCertificateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GenerateClientCertificateRequestFilterSensitiveLog = GenerateClientCertificateRequestFilterSensitiveLog;
const GetAccountRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetAccountRequestFilterSensitiveLog = GetAccountRequestFilterSensitiveLog;
const GetApiKeyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetApiKeyRequestFilterSensitiveLog = GetApiKeyRequestFilterSensitiveLog;
const GetApiKeysRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetApiKeysRequestFilterSensitiveLog = GetApiKeysRequestFilterSensitiveLog;
const GetAuthorizerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetAuthorizerRequestFilterSensitiveLog = GetAuthorizerRequestFilterSensitiveLog;
const GetAuthorizersRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetAuthorizersRequestFilterSensitiveLog = GetAuthorizersRequestFilterSensitiveLog;
const GetBasePathMappingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetBasePathMappingRequestFilterSensitiveLog = GetBasePathMappingRequestFilterSensitiveLog;
const BasePathMappingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BasePathMappingsFilterSensitiveLog = BasePathMappingsFilterSensitiveLog;
const GetBasePathMappingsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetBasePathMappingsRequestFilterSensitiveLog = GetBasePathMappingsRequestFilterSensitiveLog;
const GetClientCertificateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetClientCertificateRequestFilterSensitiveLog = GetClientCertificateRequestFilterSensitiveLog;
const ClientCertificatesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ClientCertificatesFilterSensitiveLog = ClientCertificatesFilterSensitiveLog;
const GetClientCertificatesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetClientCertificatesRequestFilterSensitiveLog = GetClientCertificatesRequestFilterSensitiveLog;
const GetDeploymentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDeploymentRequestFilterSensitiveLog = GetDeploymentRequestFilterSensitiveLog;
const DeploymentsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeploymentsFilterSensitiveLog = DeploymentsFilterSensitiveLog;
const GetDeploymentsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDeploymentsRequestFilterSensitiveLog = GetDeploymentsRequestFilterSensitiveLog;
const GetDocumentationPartRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDocumentationPartRequestFilterSensitiveLog = GetDocumentationPartRequestFilterSensitiveLog;
const DocumentationPartsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DocumentationPartsFilterSensitiveLog = DocumentationPartsFilterSensitiveLog;
const GetDocumentationPartsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDocumentationPartsRequestFilterSensitiveLog = GetDocumentationPartsRequestFilterSensitiveLog;
const GetDocumentationVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDocumentationVersionRequestFilterSensitiveLog = GetDocumentationVersionRequestFilterSensitiveLog;
const DocumentationVersionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DocumentationVersionsFilterSensitiveLog = DocumentationVersionsFilterSensitiveLog;
const GetDocumentationVersionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDocumentationVersionsRequestFilterSensitiveLog = GetDocumentationVersionsRequestFilterSensitiveLog;
const GetDomainNameRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDomainNameRequestFilterSensitiveLog = GetDomainNameRequestFilterSensitiveLog;
const DomainNamesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DomainNamesFilterSensitiveLog = DomainNamesFilterSensitiveLog;
const GetDomainNamesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDomainNamesRequestFilterSensitiveLog = GetDomainNamesRequestFilterSensitiveLog;
const ExportResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ExportResponseFilterSensitiveLog = ExportResponseFilterSensitiveLog;
const GetExportRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetExportRequestFilterSensitiveLog = GetExportRequestFilterSensitiveLog;
const GatewayResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GatewayResponseFilterSensitiveLog = GatewayResponseFilterSensitiveLog;
const GetGatewayResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetGatewayResponseRequestFilterSensitiveLog = GetGatewayResponseRequestFilterSensitiveLog;
const GatewayResponsesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GatewayResponsesFilterSensitiveLog = GatewayResponsesFilterSensitiveLog;
const GetGatewayResponsesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetGatewayResponsesRequestFilterSensitiveLog = GetGatewayResponsesRequestFilterSensitiveLog;
const GetIntegrationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetIntegrationRequestFilterSensitiveLog = GetIntegrationRequestFilterSensitiveLog;
const GetIntegrationResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetIntegrationResponseRequestFilterSensitiveLog = GetIntegrationResponseRequestFilterSensitiveLog;
const GetMethodRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMethodRequestFilterSensitiveLog = GetMethodRequestFilterSensitiveLog;
const GetMethodResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMethodResponseRequestFilterSensitiveLog = GetMethodResponseRequestFilterSensitiveLog;
const GetModelRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetModelRequestFilterSensitiveLog = GetModelRequestFilterSensitiveLog;
const GetModelsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetModelsRequestFilterSensitiveLog = GetModelsRequestFilterSensitiveLog;
const ModelsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ModelsFilterSensitiveLog = ModelsFilterSensitiveLog;
const GetModelTemplateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetModelTemplateRequestFilterSensitiveLog = GetModelTemplateRequestFilterSensitiveLog;
const TemplateFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TemplateFilterSensitiveLog = TemplateFilterSensitiveLog;
const GetRequestValidatorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetRequestValidatorRequestFilterSensitiveLog = GetRequestValidatorRequestFilterSensitiveLog;
const GetRequestValidatorsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetRequestValidatorsRequestFilterSensitiveLog = GetRequestValidatorsRequestFilterSensitiveLog;
const RequestValidatorsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RequestValidatorsFilterSensitiveLog = RequestValidatorsFilterSensitiveLog;
const GetResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetResourceRequestFilterSensitiveLog = GetResourceRequestFilterSensitiveLog;
const GetResourcesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetResourcesRequestFilterSensitiveLog = GetResourcesRequestFilterSensitiveLog;
const ResourcesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ResourcesFilterSensitiveLog = ResourcesFilterSensitiveLog;
const GetRestApiRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetRestApiRequestFilterSensitiveLog = GetRestApiRequestFilterSensitiveLog;
const GetRestApisRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetRestApisRequestFilterSensitiveLog = GetRestApisRequestFilterSensitiveLog;
const RestApisFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RestApisFilterSensitiveLog = RestApisFilterSensitiveLog;
const GetSdkRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSdkRequestFilterSensitiveLog = GetSdkRequestFilterSensitiveLog;
const SdkResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SdkResponseFilterSensitiveLog = SdkResponseFilterSensitiveLog;
const GetSdkTypeRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSdkTypeRequestFilterSensitiveLog = GetSdkTypeRequestFilterSensitiveLog;
const SdkConfigurationPropertyFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SdkConfigurationPropertyFilterSensitiveLog = SdkConfigurationPropertyFilterSensitiveLog;
const SdkTypeFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SdkTypeFilterSensitiveLog = SdkTypeFilterSensitiveLog;
const GetSdkTypesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSdkTypesRequestFilterSensitiveLog = GetSdkTypesRequestFilterSensitiveLog;
const SdkTypesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SdkTypesFilterSensitiveLog = SdkTypesFilterSensitiveLog;
const GetStageRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetStageRequestFilterSensitiveLog = GetStageRequestFilterSensitiveLog;
const GetStagesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetStagesRequestFilterSensitiveLog = GetStagesRequestFilterSensitiveLog;
const StagesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StagesFilterSensitiveLog = StagesFilterSensitiveLog;
const GetTagsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTagsRequestFilterSensitiveLog = GetTagsRequestFilterSensitiveLog;
const TagsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagsFilterSensitiveLog = TagsFilterSensitiveLog;
const GetUsageRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUsageRequestFilterSensitiveLog = GetUsageRequestFilterSensitiveLog;
const UsageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UsageFilterSensitiveLog = UsageFilterSensitiveLog;
const GetUsagePlanRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUsagePlanRequestFilterSensitiveLog = GetUsagePlanRequestFilterSensitiveLog;
const GetUsagePlanKeyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUsagePlanKeyRequestFilterSensitiveLog = GetUsagePlanKeyRequestFilterSensitiveLog;
const GetUsagePlanKeysRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUsagePlanKeysRequestFilterSensitiveLog = GetUsagePlanKeysRequestFilterSensitiveLog;
const UsagePlanKeysFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UsagePlanKeysFilterSensitiveLog = UsagePlanKeysFilterSensitiveLog;
const GetUsagePlansRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUsagePlansRequestFilterSensitiveLog = GetUsagePlansRequestFilterSensitiveLog;
const UsagePlansFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UsagePlansFilterSensitiveLog = UsagePlansFilterSensitiveLog;
const GetVpcLinkRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetVpcLinkRequestFilterSensitiveLog = GetVpcLinkRequestFilterSensitiveLog;
const GetVpcLinksRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetVpcLinksRequestFilterSensitiveLog = GetVpcLinksRequestFilterSensitiveLog;
const VpcLinksFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.VpcLinksFilterSensitiveLog = VpcLinksFilterSensitiveLog;
const ImportApiKeysRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImportApiKeysRequestFilterSensitiveLog = ImportApiKeysRequestFilterSensitiveLog;
const DocumentationPartIdsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DocumentationPartIdsFilterSensitiveLog = DocumentationPartIdsFilterSensitiveLog;
const ImportDocumentationPartsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImportDocumentationPartsRequestFilterSensitiveLog = ImportDocumentationPartsRequestFilterSensitiveLog;
const ImportRestApiRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImportRestApiRequestFilterSensitiveLog = ImportRestApiRequestFilterSensitiveLog;
const PutGatewayResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutGatewayResponseRequestFilterSensitiveLog = PutGatewayResponseRequestFilterSensitiveLog;
const PutIntegrationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutIntegrationRequestFilterSensitiveLog = PutIntegrationRequestFilterSensitiveLog;
const PutIntegrationResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutIntegrationResponseRequestFilterSensitiveLog = PutIntegrationResponseRequestFilterSensitiveLog;
const PutMethodRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutMethodRequestFilterSensitiveLog = PutMethodRequestFilterSensitiveLog;
const PutMethodResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutMethodResponseRequestFilterSensitiveLog = PutMethodResponseRequestFilterSensitiveLog;
const PutRestApiRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutRestApiRequestFilterSensitiveLog = PutRestApiRequestFilterSensitiveLog;
const TagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagResourceRequestFilterSensitiveLog = TagResourceRequestFilterSensitiveLog;
const TestInvokeAuthorizerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TestInvokeAuthorizerRequestFilterSensitiveLog = TestInvokeAuthorizerRequestFilterSensitiveLog;
const TestInvokeAuthorizerResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TestInvokeAuthorizerResponseFilterSensitiveLog = TestInvokeAuthorizerResponseFilterSensitiveLog;
const TestInvokeMethodRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TestInvokeMethodRequestFilterSensitiveLog = TestInvokeMethodRequestFilterSensitiveLog;
const TestInvokeMethodResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TestInvokeMethodResponseFilterSensitiveLog = TestInvokeMethodResponseFilterSensitiveLog;
const UntagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UntagResourceRequestFilterSensitiveLog = UntagResourceRequestFilterSensitiveLog;
const PatchOperationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PatchOperationFilterSensitiveLog = PatchOperationFilterSensitiveLog;
const UpdateAccountRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateAccountRequestFilterSensitiveLog = UpdateAccountRequestFilterSensitiveLog;
const UpdateApiKeyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateApiKeyRequestFilterSensitiveLog = UpdateApiKeyRequestFilterSensitiveLog;
const UpdateAuthorizerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateAuthorizerRequestFilterSensitiveLog = UpdateAuthorizerRequestFilterSensitiveLog;
const UpdateBasePathMappingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateBasePathMappingRequestFilterSensitiveLog = UpdateBasePathMappingRequestFilterSensitiveLog;
const UpdateClientCertificateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateClientCertificateRequestFilterSensitiveLog = UpdateClientCertificateRequestFilterSensitiveLog;
const UpdateDeploymentRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateDeploymentRequestFilterSensitiveLog = UpdateDeploymentRequestFilterSensitiveLog;
const UpdateDocumentationPartRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateDocumentationPartRequestFilterSensitiveLog = UpdateDocumentationPartRequestFilterSensitiveLog;
const UpdateDocumentationVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateDocumentationVersionRequestFilterSensitiveLog = UpdateDocumentationVersionRequestFilterSensitiveLog;
const UpdateDomainNameRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateDomainNameRequestFilterSensitiveLog = UpdateDomainNameRequestFilterSensitiveLog;
const UpdateGatewayResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateGatewayResponseRequestFilterSensitiveLog = UpdateGatewayResponseRequestFilterSensitiveLog;
const UpdateIntegrationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateIntegrationRequestFilterSensitiveLog = UpdateIntegrationRequestFilterSensitiveLog;
const UpdateIntegrationResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateIntegrationResponseRequestFilterSensitiveLog = UpdateIntegrationResponseRequestFilterSensitiveLog;
const UpdateMethodRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateMethodRequestFilterSensitiveLog = UpdateMethodRequestFilterSensitiveLog;
const UpdateMethodResponseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateMethodResponseRequestFilterSensitiveLog = UpdateMethodResponseRequestFilterSensitiveLog;
const UpdateModelRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateModelRequestFilterSensitiveLog = UpdateModelRequestFilterSensitiveLog;
const UpdateRequestValidatorRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateRequestValidatorRequestFilterSensitiveLog = UpdateRequestValidatorRequestFilterSensitiveLog;
const UpdateResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateResourceRequestFilterSensitiveLog = UpdateResourceRequestFilterSensitiveLog;
const UpdateRestApiRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateRestApiRequestFilterSensitiveLog = UpdateRestApiRequestFilterSensitiveLog;
const UpdateStageRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateStageRequestFilterSensitiveLog = UpdateStageRequestFilterSensitiveLog;
const UpdateUsageRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateUsageRequestFilterSensitiveLog = UpdateUsageRequestFilterSensitiveLog;
const UpdateUsagePlanRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateUsagePlanRequestFilterSensitiveLog = UpdateUsagePlanRequestFilterSensitiveLog;
const UpdateVpcLinkRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateVpcLinkRequestFilterSensitiveLog = UpdateVpcLinkRequestFilterSensitiveLog;
